src/vulkan/vulkansetup.cpp

```diff
@@ -2,260 +2,44 @@
 
 #include "vulkanutils.h"
 
-#include <cstring>
 #include <iostream>
 #include <string>
 
-#if defined(__APPLE__) || defined(__linux)
-#include <dlfcn.h>
-#endif
-
-#if defined (_WIN32)
-constexpr char* const VK_LIB_NAME = "vulkan-1.dll";
-#elif defined (__APPLE__)
-constexpr char *const VK_LIB_NAME = "libvulkan.1.dylib";
-#elif defined (__linux)
-constexpr char* const VK_LIB_NAME = "libvulkan.1.so";
-#else
-#error "The DireWolf renderer is not yet setup for Vulkan on this platform. Supported operating systems are Linux and Windows";
-#endif
-
-namespace {
-#if defined(_WIN32)
-HMODULE s_vulkan_library;
-#else
-void *s_vulkan_library;
-#endif
-}
-
 namespace dw::vulkan {
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////
 
-// TODO: should probably take some init-struct
-TempVulkanSetupObject::TempVulkanSetupObject(std::vector<const char*>* desiredExtensions /* = nullptr */)
-: m_isValid(false)
-, m_instance(VK_NULL_HANDLE)
-{
-#if defined( DW_VERBOSE_LOG_VK )
-    std::cout << "\nTemporaryVulkanSetupObject ctor\n";
-#endif
-    m_isValid = initialize(desiredExtensions);
-}
-
-////////////////////////////////////////////////////////////////////////////////////////////////////
-
-bool TempVulkanSetupObject::initialize(std::vector<const char*>* desiredExtensions)
+bool InitializeVulkan()
 {
-    if (m_isValid) {
-        std::cerr << "Trying to initialize TempVulkanSetupObject after it has already been initialized!" << std::endl;
+    // Get run-time library and initialize global level functions
+    VulkanRTLPtr vulkanRTL = GetRuntimeLibs();
+    if (!vulkanRTL || !InitProcAddress(vulkanRTL) || !InitGlobalLevelFunction()) {
```

Comment:
Handle error / message ?

Reply:
I think the functions themselves print something if they fail.. gonna look into it and make sure that they do